wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,491 @@
+
+      const balance = balances.find(
+        ({ denom }) => denom === ibcAsset.deposit.denom,
+      );
+      setRemoteChainBalance(balance ? BigInt(balance.amount) : 0n);
+    };
+
+    void doQuery();
+
+    return () => {
+      isCancelled = true;
+    };
+  }, [remoteChainAddress, isRemoteChainAddressInvalid, ibcAsset]);
+
+  const handleAmountChange = e => {
+    setAmount(e.target.value);
+  };
+
+  const fillFromKeplr = async () => {
+    assert(ibcAsset);
+    // @ts-expect-error window keys
+    const { keplr } = window;
+    const offlineSigner = await keplr.getOfflineSignerOnlyAmino(
+      ibcAsset.chainInfo.chainId,
+    );
+    const accounts = await offlineSigner.getAccounts();
+    setRemoteChainAddress(accounts[0].address);
+    setRemoteChainSigner(offlineSigner);
+  };
+
+  const isAmountInvalid = useMemo(() => {
+    switch (direction) {
+      case IbcDirection.Withdrawal:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (purseBalance ?? 0n);
+        } catch {
+          return true;
+        }
+      case IbcDirection.Deposit:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (remoteChainBalance ?? 0n);
+        } catch {
+          return true;
+        }
+    }
+  }, [amount, purse, purseBalance]);
+
+  const close = () => {
+    setAmount('');
+    setRemoteChainAddress('');
+    setRemoteChainSigner(null);
+    setError('');
+    handleClose();
+  };
+
+  const send = async () => {
+    setError('');
+
+    let val;
+    try {
+      val = String(
+        parseAsValue(
+          amount,
+          purse?.displayInfo?.assetKind,
+          purse?.displayInfo?.decimalPlaces,
+        ),
+      );
+    } catch (e) {
+      setError(String(e));
+    }
+
+    assert(ibcAsset);
+
+    if (direction === IbcDirection.Deposit) {
+      assert(remoteChainSigner && keplrConnection);
+      const { chainInfo, deposit } = ibcAsset;
+      const { sourceChannel, sourcePort, denom, gas } = deposit;
+      assert(gas);
+
+      setInProgress(true);
+      const signer = await SigningStargateClient.connectWithSigner(
+        chainInfo.rpc,
+        remoteChainSigner,
+      );
+      try {
+        const res = await signer.sendIbcTokens(
+          remoteChainAddress,
+          keplrConnection.address,
+          {
+            amount: val,
+            denom: denom,
+          },
+          sourcePort,
+          sourceChannel,
+          undefined,
+          timeoutTimestampSeconds(),
+          {
+            amount: [{ amount: '0', denom }],
+            gas,
+          },
+        );
+        close();
+        showSnackbar(
+          <>
+            Successfully executed transaction{' '}
```

Comment:
Looking at [sendIbcTokens](https://github.com/cosmos/cosmjs/blob/33271bc51c/packages/stargate/src/signingstargateclient.ts#L246), it doesn't seem to do `assertIsDeliverTxSuccess(tx)` so I think we need to do that before declaring success.

Reply:
Done